Kickstarter-iOS/Views/LoadingButton.swift

```diff
@@ -0,0 +1,115 @@
+import Library
+import Prelude
+import UIKit
+
+final class LoadingButton: UIButton {
+  // MARK: - Properties
+
+  private lazy var activityIndicator: UIActivityIndicatorView = {
+    UIActivityIndicatorView(style: .white)
+      |> \.hidesWhenStopped .~ true
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  public var isLoading: Bool = false {
+    didSet {
+      guard self.isLoading != oldValue else { return }
+
+      if self.isLoading {
+        self.startLoading()
+      } else {
+        self.stopLoading()
+      }
+
+      _ = self
+        |> \.isUserInteractionEnabled .~ !self.isLoading
+    }
+  }
+
+  private var originalTitles: [UInt: String] = [:]
```

Comment:
Is there a reason you chose to do this without a view model?

Reply:
I simply tried to go with the simplest approach but can back this by a VM to be more consistent with the rest of the codebase 👍